Very first draft of mapping script + raw data

LienReyserhove opened the pull request.

LienReyserhove committed:
Add rownames

LienReyserhove committed:
Map scientificName

LienReyserhove committed:
Map scientificName pt 2

LienReyserhove committed:
Define first settings

LienReyserhove committed:
complete README + some examples pt1

LienReyserhove committed:
Add multipele distributions per taxon to template file

LienReyserhove committed:
Add library inborutils

LienReyserhove committed:
Generate taxonID

LienReyserhove committed:
Add classification information pt1

LienReyserhove committed:
check for duplicates step

LienReyserhove committed:
Remove duplicated rows

LienReyserhove committed:
Switch places mapping

LienReyserhove committed:
Add taxon core mapping

LienReyserhove committed:
Add comments

LienReyserhove committed:
remove extra libraries

LienReyserhove committed:
Finish taxon mapping

LienReyserhove committed:
Add distribution mapping

LienReyserhove committed:
Delete kingdom & taxonRank from data template

LienReyserhove committed:
Undo remove kingdom

LienReyserhove committed:
Change taxon Core information

change taxonID (integrate kingdom information)
remove references
remove phylum/class/order/family

LienReyserhove committed:
Clean input data

LienReyserhove committed:
Remove species name match step

LienReyserhove committed:
Add taxonRank information using nameparser

LienReyserhove committed:
Add additional explanations

LienReyserhove committed:
Remove `Flanders` for one record

LienReyserhove committed:
Add distribution mapping

LienReyserhove committed:
Add taxon rank to raw data file

I think it's better to add this term to the raw data file as the GBIF nameparser sometimes renders errors (see https://github.com/gbif/name-parser/issues/26)

LienReyserhove committed:
Rearrange structure

peterdesmet commented:
Suggested updates to structure:

1. Setup
2. Read raw data
3. Preprocess raw data
    - Data structure
        - Remove empty rows
        - Clean col names
        - Prepend with raw
        - Preview data
    - Scientific names
        - Parse scientific names (and return non-scientific names)
        - Generate taxonID
        - Preview data: number of taxa per kingdom + number of distributions
4. Create taxon core (start from non-duplicated df)
5. Create distribution extension

peterdesmet commented:
@LienReyserhove other remarks:

- vectorize: keep close to taxonID code block
- Don't use `View`: it doesn't knit
- Removing double spaces from any field could be part of pre-processing, right after remove empty rows
- Use `rgbif:parsenames` to make super clear that function comes from that package
- Please remove all Dutch remarks

LienReyserhove committed:
Remove dutch comments

LienReyserhove committed:
Remove UTF-8 encoding + install.packages step

LienReyserhove committed:
Change taxonID mapping

Shift vdigest function and taxonID mapping backwards

LienReyserhove committed:
Change introduction section

LienReyserhove committed:
Add nameparser function and rankmarker information

LienReyserhove committed:
Shift removal of duplicated taxa forward

LienReyserhove committed:
Add random information to record-level terms

LienReyserhove committed:
Change taxonRank mapping

LienReyserhove committed:
Add information to terms specific for taxon core

LienReyserhove committed:
Remove taxon_rank column + add new taxa for nameparser function

LienReyserhove committed:
Rerun code to generate taxon core

LienReyserhove commented:
I integrated the suggestions in the following commits.
Remark: I added the taxon rank information in the section `scientific names`. This is needed to make a good preview af the data

LienReyserhove committed:
Add preview of the data

Probably way too complicated... + taxon rank needs some cleaning

peterdesmet committed:
Review and update section 1-3

LienReyserhove committed:
Remove taxon_rank information from README in raw_data

peterdesmet committed:
Use input_ rather than raw_ as prefix

The problem with `raw_` is that it implies unchanged, while we  actually do some processing on the data before mapping. input_ is more neutral

peterdesmet committed:
Preview data before and after processing

peterdesmet committed:
Add totals to taxon rank preview with adorn_totals()

peterdesmet committed:
Use simplified distinct

peterdesmet committed:
Review taxon core mapping

- Use my_... as dummy values
- Inspect kingdom values
- Inspect taxonrank values

peterdesmet committed:
Review distribution mapping

- Turn around case_when conditions
- Inspect all unique values where useful
- Add links to vocabs

peterdesmet committed:
Use tidyverse distinct() + pull() rather than unique()

peterdesmet committed:
Run code on data

peterdesmet committed:
Add build option to src.Rproj

peterdesmet committed:
Merge branch 'create-data-template' of https://github.com/trias-project/checklist-recipe into create-data-template

peterdesmet committed:
Ignore html files in src
